pkg/rancher-desktop/main/credentialServer/__tests__/credentialUtils.spec.ts

```diff
@@ -0,0 +1,162 @@
+
+      expect(command).toEqual('docker-credential-pikachu');
+      expect(args).toEqual(['pika']);
+      expect(options).toMatchObject({
+        env:   { PATH: expect.stringContaining(resourcesPath) },
+        stdio: [expect.anything(), 'pipe', expect.anything()],
+      });
+
+      return Promise.resolve({ stdout: expected }) as any;
+    });
+    await expect(runCommand('pika')).resolves.toEqual(expected);
+  });
+
+  it('errors out on failing to read config', async() => {
+    const error = new Error('Some error');
+
+    jest.spyOn(fs.promises, 'readFile').mockImplementation((filepath) => {
+      const home = findHomeDir() ?? '';
+
+      expect(filepath).toEqual(path.join(home, '.docker', 'config.json'));
+
+      return Promise.reject(error);
+    });
+
+    jest.mocked(spawnFile).mockImplementation(() => Promise.resolve({}));
+
+    await expect(runCommand('pika')).rejects.toBe(error);
+    expect(jest.mocked(spawnFile)).not.toHaveBeenCalled();
+  });
+
+  describe.each([
+    {
+      description: 'overridden', host: 'override.test', executable: 'bulbasaur',
+    },
+    {
+      description: 'not overridden', host: 'default.test', executable: 'pikachu',
+    },
+  ])('helper $description', ({ host, executable }) => {
+    beforeEach(() => {
+      jest.spyOn(fs.promises, 'readFile').mockImplementation((filepath) => {
+        const home = findHomeDir() ?? '';
+
+        expect(filepath).toEqual(path.join(home, '.docker', 'config.json'));
+
+        return Promise.resolve(JSON.stringify({
+          credsStore:  'pikachu',
+          credHelpers: { 'override.test': 'bulbasaur' },
+        }));
+      });
+    });
+
+    test.each([
+      { command: 'get', input: host },
+      { command: 'erase', input: host },
+      { command: 'store', input: JSON.stringify({ ServerURL: host, arg: 'x' }) },
+      {
+        command: 'unknown command', input: host, override: 'pikachu',
+      },
+    ])('on $command', async({ command, input, override }) => {
+      const expected = 'password';
+
+      jest.mocked(spawnFile).mockImplementation((file, args, options) => {
+        expect(file).toEqual(`docker-credential-${ override ?? executable }`);
+        expect(args).toEqual([command]);
+        expect(options).toMatchObject({ stdio: [expect.any(stream.Readable), expect.anything(), expect.anything()] });
+
+        return Promise.resolve({ stdout: expected }) as any;
+      });
+
+      await expect(runCommand(command, input)).resolves.toEqual(expected);
+    });
+  });
+});
+
+describe('list', () => {
+  let config: { credsStore: string, credHelpers?: Record<string, string>} = { credsStore: 'unset' };
+  let helpers: Record<string, any> = {};
+
+  beforeEach(() => {
+    jest.spyOn(fs.promises, 'readFile').mockImplementation((filepath, options) => {
+      const home = findHomeDir() ?? '';
+
+      expect(filepath).toEqual(path.join(home, '.docker', 'config.json'));
+      expect(options).toMatchObject({ encoding: 'utf-8' });
```

Comment:
Again I'm not sure what checking the options field adds to this test. Maybe if we were handling `ucs-2` on windows...

Reply:
Also removed.